python/lsst/ts/mtrotator/rotator_csc.py

```diff
@@ -519,13 +548,134 @@ async def do_trackStart(self, data: salobj.BaseMsgType) -> None:
         Once this is run you must issue ``track`` commands at 10-20Hz
         until you are done tracking, then issue the ``stop`` command.
         """
+
+        self.assert_is_motion_locked()
+
         self.assert_enabled_substate(EnabledSubstate.STATIONARY)
         await self.run_command(
             code=enums.CommandCode.SET_ENABLED_SUBSTATE,
             param1=enums.SetEnabledSubstateParam.TRACK,
         )
         self._tracking_started_telemetry_counter = 2
 
+    async def do_lockMotion(self, data: salobj.BaseMsgType) -> None:
+        """Lock the rotator motion.
+
+        Parameters
+        ----------
+        data : `salobj.BaseMsgType`
+            Data of the SAL message.
+
+        Raises
+        ------
+        `salobj.ExpectedError`
+            When the motion is failed to stop.
+        """
+
+        if self._is_motion_locked:
+            return
+
+        await self.cmd_lockMotion.ack_in_progress(data, timeout=self.COMMAND_TIMEOUT)
+
+        await self._pub_event_motion_lock_state(
+            1, data.private_identity
+        )  # Use MotionLockState.LOCKING
+
+        # Workaround the mypy check
+        assert self.client is not None
+
+        # Stop the motion when it is not stationary
```

Comment:
You might want to take some defensive actions here in the various places that an error might happen. I noticed that, if an error happen between here a line 613 below, the system will be left in the `LOCKING` state. 

For example, in line 605, you raise an exception. That would make the command to lock fail and leave the system in `LOCKING` state, and probably in a weird transient state. I think you might have to consider Faulting the system (in addition to raising the exception).

Reply:
Improved in the commit: [b4e39d1](https://github.com/lsst-ts/ts_mtrotator/pull/97/commits/b4e39d11dc643201ed9fc28fd9b89b0c45c65069). Thanks!